src/pages/appointments/create.tsx

```diff
@@ -0,0 +1,137 @@
+import {
+  Button,
+  Card,
+  Elevation,
+  FormGroup,
+  Icon,
+  MenuItem,
+} from "@blueprintjs/core";
+import { TimePicker } from "@blueprintjs/datetime";
+import { DateInput2 } from "@blueprintjs/datetime2";
+import { Select2 } from "@blueprintjs/select";
+import { format, parse } from "date-fns";
+import { NextPage } from "next";
+import { useRouter } from "next/router";
+import { useCallback, useState } from "react";
+import { useDispatch } from "react-redux";
+import AppointmentTypes from "src/redux/types/appointmentTypes";
+import styles from "../../styles/pages/appointments/Create.module.css";
+
+const Create: NextPage = () => {
+  const [dateValue, setDateValue] = useState<string | null>(null);
+  const [startTime, setStartTime] = useState<Date>(
+    new Date("2020-01-01T00:00:00")
+  );
+  const [endTime, setEndTime] = useState<Date>(new Date("2020-01-01T00:00:00"));
+  const [serviceType, setServiceType] = useState<string>("Canned");
+
+  const dispatch = useDispatch();
+  const router = useRouter();
+
+  const handleRequestSubmit = (): void => {
+    const service_type = serviceType.toUpperCase();
+    const start_time = new Date(
+      dateValue + "T" + format(startTime, timeFnsFormat)
+    ).toString();
+    const end_time = new Date(
+      dateValue + "T" + format(endTime, timeFnsFormat)
+    ).toString();
+
+    console.log(
+      "Payload: " + service_type + ", " + start_time + ", " + end_time
+    );
+
+    dispatch({
+      type: AppointmentTypes.CREATE_APPOINTMENT,
+      payload: { service_type, start_time, end_time },
+    });
+  };
+
+  const handleDateChange = useCallback(setDateValue, []);
+  const dateFnsFormat = "PPPP";
```

Comment:
Can we define these constants that don't change outside the functional component?